manifests/init.pp

```diff
@@ -188,98 +188,98 @@
+  Boolean $ssl_management_fail_if_no_peer_cert                                                     = $rabbitmq::params::ssl_management_fail_if_no_peer_cert,
+  Optional[Array] $ssl_versions                                                                    = undef,
+  Boolean $ssl_secure_renegotiate                                                                  = $rabbitmq::params::ssl_secure_renegotiate,
+  Boolean $ssl_reuse_sessions                                                                      = $rabbitmq::params::ssl_reuse_sessions,
+  Boolean $ssl_honor_cipher_order                                                                  = $rabbitmq::params::ssl_honor_cipher_order,
+  Optional[Stdlib::Absolutepath] $ssl_dhfile                                                       = undef,
+  Array $ssl_ciphers                                                                               = $rabbitmq::params::ssl_ciphers,
+  Boolean $stomp_ensure                                                                            = $rabbitmq::params::stomp_ensure,
+  Boolean $ldap_auth                                                                               = $rabbitmq::params::ldap_auth,
+  String $ldap_server                                                                              = $rabbitmq::params::ldap_server,
+  Optional[String] $ldap_user_dn_pattern                                                           = $rabbitmq::params::ldap_user_dn_pattern,
+  String $ldap_other_bind                                                                          = $rabbitmq::params::ldap_other_bind,
+  Boolean $ldap_use_ssl                                                                            = $rabbitmq::params::ldap_use_ssl,
+  Integer[1, 65535] $ldap_port                                                                     = $rabbitmq::params::ldap_port,
+  Boolean $ldap_log                                                                                = $rabbitmq::params::ldap_log,
+  Hash $ldap_config_variables                                                                      = $rabbitmq::params::ldap_config_variables,
+  Integer[1, 65535] $stomp_port                                                                    = $rabbitmq::params::stomp_port,
+  Boolean $stomp_ssl_only                                                                          = $rabbitmq::params::stomp_ssl_only,
+  Boolean $wipe_db_on_cookie_change                                                                = $rabbitmq::params::wipe_db_on_cookie_change,
+  String $cluster_partition_handling                                                               = $rabbitmq::params::cluster_partition_handling,
+  Variant[Integer[-1],Enum['unlimited'],Pattern[/^(infinity|\d+(:(infinity|\d+))?)$/]] $file_limit = $rabbitmq::params::file_limit,
```

Comment:
This is the reason why I don't like the `=` alignment ;) I do wonder if puppet-systemd should have a type for this so you can reuse it in cases like these. Currently it's hidden in a struct so no reuse there.

Reply:
I think a reuse is possible, if we would use the type correctly.

```puppet
  if $facts['systemd'] { # systemd fact provided by systemd module
    systemd::service_limits { 'rabbitmq.service':
      limits          => {'LimitNOFILE' => $file_limit},
      # The service will be notified when config changes
      restart_service => false,
    }   
  }
```

We could reuse the struct if we would allow people to pass the whole hash. What we could do: Create a new optional parameter with the struct from puppet-systemd. If it is set we pass it to ```$systemd::service_limits::limits```, otherwise we use ```$rabbitmq::file_limit```. This would be easier if ```$rabbitmq::file_limit``` would not be used for legacy ```/etc/limits``` entries as well.

Reply:
Re: Alignment, it is way easier to read if you ignore whitespace changes in the diff.